app/service/elasticsearch_service.py

```diff
@@ -20,17 +20,33 @@ def __init__(self, cloud_id: str, user_id: str, secret_id: str, index: str):
 
     def execute_search_query(self, search_query: SearchQuery):
         result = self.elasticsearch.search(
-            index="most-tech-dev",
+            index=self.index,
+            search_type="dfs_query_then_fetch",
             body={
                 "query": {
-                    "multi_match": {
-                        "query": f"{search_query.keystroke}",
-                        "fuzziness": "AUTO",
-                        "prefix_length": 1,
-                        "type": "bool_prefix",
-                        "fields": ["keystroke", "keystroke._2gram", "keystroke._3gram"],
-                    }
-                }
-            },
-        )["hits"]["hits"]
-        return [pattern["_source"] for pattern in result]
+                    "bool": {
+                        "must": [
+                            {
+                                "multi_match": {
+                                    "query": f"{search_query.paragraph}",
+                                    "fuzziness": "AUTO",
+                                    "prefix_length": 1,
+                                    "type": "bool_prefix",
+                                    "fields": [
+                                        "paragraph",
+                                        "paragraph._2gram",
+                                        "paragraph._3gram",
+                                    ],
+                                },
+                            }
+                        ],
+                        "should": [
```

Comment:
What's the difference between should and must? If we add another `should` to the `bool` query there is a posiibility that labels won't be taken into account in search even if selected in ui. Labels are mandatory so let's chage to `must`

Reply:
if it is must it can't find anything when the label is "". If you are searching with must you must define both paragraph and labels. So it won't work with initial query when we do not use label.

Reply:
Can you try `"zero_terms_query": "all"`?